screens/ReservationCalendarScreen.js

```diff
@@ -0,0 +1,508 @@
+import {CheckBox} from 'react-native-elements'
+import Icon from 'react-native-vector-icons/Ionicons'
+import {CalendarEvent} from "../components/CalendarEvent";
+import {Ionicons} from '@expo/vector-icons';
+import {compose} from "redux";
+import {connect} from 'react-redux'
+import {MaterialCommunityIcons} from '@expo/vector-icons';
+import {getTableLayouts, getTableLayout} from '../actions'
+
+class CalendarScreen extends React.Component {
+    static navigationOptions = {
+        header: null
+    }
+    static contextType = LocaleContext
+
+    constructor(props, context) {
+        super(props, context)
+        const timezone = TimeZoneService.getTimeZone()
+        this.state = {
+            isLoading: true,
+            calendarMode: 'month',
+            selectedDate: moment(new Date()).tz(timezone).format("YYYY-MM-DD"),
+            users: [],
+            isShowModal: false,
+            calendarTypeIndex: 0,
+            searchTypeIndex: 1,
+            modalTasks: [],
+            nowYear: new Date().getFullYear(),
+            nowMonth: new Date().getMonth() + 1,
+            monthLoading: false,
+            reservationEvents: [],
+            reservationEventsByDate: [],
+            showReservationFormModal: false,
+        }
+        context.localize({
+            en: {
+                calendar: {
+                    monthly: 'Monthly',
+                    daily: 'Daily',
+                    reservation: 'Reservation',
+                    calendarOption: 'Calendar Option',
+                }
+            },
+            zh: {
+                calendar: {
+                    monthly: '月曆',
+                    daily: '日曆',
+                    reservation: '訂位',
+                    calendarOption: '日程表選項',
+                }
+            }
+        })
+        LocaleConfig.locales['zh-Hant-TW'] = {
+            monthNames: ['01', '02', '03', '04', '05', '06', '07', '08', '09', '10', '11', '12'],
+            monthNamesShort: ['01', '02', '03', '04', '05', '06', '07', '08', '09', '10', '11', '12'],
+            dayNames: ['週日', '週一', '週二', '週三', '週四', '週五', '週六'],
+            dayNamesShort: ['週日', '週一', '週二', '週三', '週四', '週五', '週六'],
+            today: '今日'
+        }
+        if (this.context?.locale === 'zh-Hant-TW' || this.context?.locale === 'zh-TW')
+            LocaleConfig.defaultLocale = 'zh-Hant-TW'
+    }
+
+    componentDidMount() {
+        this.props.navigation.openDrawer
+
+        this.getUsers()
+        this.props.getTableLayouts()
+        this.getReservationEvents()
+        this.getReservationEventsByDate()
+    }
+
+    refreshScreen = async () => {
+
+        this.getUsers()
+        this.props.getTableLayouts()
+        this.getReservationEvents(new Date(this.state?.selectedDate ?? new Date()).getFullYear(), moment(new Date(this.state?.selectedDate ?? new Date())).format('MM'))
+        this.getReservationEventsByDate(this.state?.selectedDate, 'BOOKED')
```

Comment:
Should I give you an API that include reservations of all states?

Reply:
Maybe an array more suitable. Like status=['BOOKED', 'CONFIRMED']